common/src/attestations/signers.rs

```diff
@@ -15,47 +19,92 @@ pub fn attestation_signers(
-
-            // Remove signers for allocations that are no longer active or recently closed
-            signers.retain(|id, _| allocations.contains_key(id));
-
-            // Create signers for new allocations
-            for (id, allocation) in allocations.iter() {
-                if !signers.contains_key(id) {
-                    let signer = AttestationSigner::new(
-                        &indexer_mnemonic,
-                        allocation,
+    let (mut signers_writer, signers_reader) =
+        Eventual::<HashMap<Address, AttestationSigner>>::new();
+
+    tokio::spawn(async move {
+        //listening to the allocation eventual and converting them to reciever
+        //using pipe for updation
+        //for temporary pupose only
+        let (allocations_tx, mut allocations_rx) =
+            watch::channel(indexer_allocations.value().await.unwrap());
+        let _p1 = indexer_allocations.pipe(move |allocatons| {
+            let _ = allocations_tx.send(allocatons);
+        });
+
+        loop {
+            select! {
+                Ok(_)= allocations_rx.changed() =>{
+                    signers_writer = modify_sigers(
+                        Arc::new(indexer_mnemonic.clone()),
                         chain_id,
-                        dispute_manager,
-                    );
-                    if let Err(e) = signer {
-                        warn!(
-                            "Failed to establish signer for allocation {}, deployment {}, createdAtEpoch {}: {}",
-                            allocation.id, allocation.subgraph_deployment.id,
-                            allocation.created_at_epoch, e
-                        );
-                    } else {
-                        signers.insert(*id, signer.unwrap());
-                    }
+                        attestation_signers_map,
+                        allocations_rx.clone(),
+                        dispute_manager_rx.clone(),
+                        signers_writer).await;
+                },
+                Ok(_)= dispute_manager_rx.changed() =>{
+                    signers_writer = modify_sigers(Arc::new(indexer_mnemonic.clone()),
+                    chain_id,
+                    attestation_signers_map,
+                    allocations_rx.clone(),
+                    dispute_manager_rx.clone(),
+                    signers_writer).await;
+                },
+                else=>{
+                    //something is wrong
+                    break;
                 }
             }
+        }
+    });
 
-            signers.clone()
+    signers_reader
+}
+async fn modify_sigers(
+    indexer_mnemonic: Arc<String>,
+    chain_id: ChainId,
+    attestation_signers_map: &'static Mutex<HashMap<Address, AttestationSigner>>,
+    allocations_rx: Receiver<HashMap<Address, Allocation>>,
+    dispute_manager_rx: Receiver<Option<Address>>,
+    mut signers_writer: EventualWriter<HashMap<Address, AttestationSigner>>,
```

Comment:
Question, is it possible for this to be `signers_writer: &mut EventualWriter<HashMap<Address, AttestationSigner>>,`?

Reply:
Thanks a lot for the detailed review, I'll update and push in the morning, am not at home as of now.

Reply:
of course, &mut would have smoothly solved the ownership issues, of signers_writer, failed to consider it first time,   I have updated the code using it.